kroxylicious-filter-test-support/src/main/java/io/kroxylicious/test/assertj/RecordBatchAssert.java

```diff
@@ -64,6 +66,101 @@ public RecordBatchAssert hasNumRecords(int expected) {
         return this;
     }
 
+    public RecordBatchAssert hasMagic(byte magic) {
+        Assertions.assertThat(actual.magic())
```

Comment:
```suggestion
        isNotNull();
        Assertions.assertThat(actual.magic())
```

Reply:
most of the values are primitives, I'll put null checks on the enums

Reply:
ah sorry I misread, that's checking the current thing is not null as a precondition. will add.